### DISCUSSION TITLE
Issues with tone() on ATTiny861

### DISCUSSION THREAD

**Discussion by AvilanHauxen** [SpenceKonde/ATTinyCore#664] [Support / Q & A]
Hi.

I have a project that I am moving from an ATTiny84 to an ATTiny861, and I am experiencing some issues when using tone().

When using a t861, it seems only part of the very first sound is auditable and everything afterwards is barely auditable and muffled.

When using the same sketch and wiring with a t84 everything sounds as expected.

I've tried using different chips and piezos and have tried with just about every pin.

The following is my testing sketch;
```c++
uint8_t piezoPin = PIN_PB3;

void setup() {
  pinMode(piezoPin, OUTPUT);
}

void loop() {
  tone(piezoPin, 1850);
  delay(100);
  noTone(piezoPin);
  delay(100);
  tone(piezoPin, 1850);
  delay(100);
  noTone(piezoPin);
  delay(100);

  delay(2000);
}
```

Arduino IDE 1.8.19
ATtinyCore 1.5.2

**Comment by SpenceKonde** [comment]
Weeeeelllll it is most definitely totally freakin busted in 2.0.0 - I don't have a piezo, but I see what I think is a manifestation of what you hear - I put it on the scope and it starts at 250 Hz if F_CPU is 8M and 500 at 15, and then jumps up to 64 times that on the second and subsequent tone. Could that explain what you're seeing? Does the frequency you pass even make ant difference? It's not for me in 2.0.0-dev

(note that the 1.x.y versions of ATTinyCore are no longer receiving any patches of an sort, all effort is directed towards 2.0.0.

And unfortunately, tinySoftSerial ALSO is not working correctly so I can't get an debug info out of it currently....

**Reply by SpenceKonde** [reply]
#665  created

**Reply by AvilanHauxen** [reply]
I have been using 1.5.2, but eventually managed to install and upload using 2.0.0, and I am experienced the same with that version.

But, I tried changing the frequencies a bit, and if I go down to 243 or lower it is no longer muffled after the first sound.
The actual sound it makes though, sounds much more like it has a frequency around 2-3000.

If the frequency is set to 244 or higher, it just makes this barely auditable buzzing sound.

I tested this for both 1.5.2 and 2.0.0 and its the same.

**Comment by SpenceKonde** [comment]
Let's take further discussion to the issue as this is now an action item for me
